### DISCUSSION TITLE
Can pages, widgets has own stores for business logic?

### DISCUSSION THREAD

**Discussion by dante01yoon** [feature-sliced/documentation#731] [Q&A] [answered]
TLDR: Should features layer be only place domain logic locate

Rather than using features layer, can we locate stores including business logic like transform request payloads, validation in pages or widget? 

There were some cases skipping widget layer, only  pages layer  remains which has components segment in there, and stores would be used in somewhere in these components. 

For me `widgets, features` layers come with more about how to `well structuring code blocks` between modules, rather than there any strict rule; `any business logic should be in features` or `large UI block should be located in widgets`.

**Comment by illright** [comment, marked as answer]
Widgets and pages certainly can have business logic and stores — there's no rule that forbids it.

However, usually, pages are not the right scope for a store. A store is usually global (used on several pages) or local to a certain component, so it might just be placed next to that component.

You can certainly start by having a model segment on a page, and maybe later move it to another layer if the project requirements need it

**Reply by dante01yoon** [reply]
Thanks!
